### PR TITLE
SEQNG-455 Modified Step and Sequence state.

### PR DESCRIPTION
**Pull request by jluhrs**
The biggest change was in the Sequence states.

### CODE REVIEW

File: modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala [hunk 1]
```diff
@@ -204,6 +204,25 @@ object Sequence {
         case Final(st, _)  => Final(st, s)
       })
 
+    def isRunning(st: State): Boolean = SequenceState.isRunning(st.status)
```
Comment (jluhrs) [inline modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala]: Added several utility functions here.

File: modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/Model.scala [hunk 2]
```diff
@@ -186,7 +187,7 @@ object Model {
     }
     case object Completed extends StepState
     case object Skipped extends StepState
-    final case class Error(msg: String) extends StepState {
+    final case class Failed(msg: String) extends StepState {
```
Comment (jluhrs) [inline modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/Model.scala]: Changed to `Failed` to be consistent.

File: modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/Model.scala [hunk 3]
```diff
@@ -243,18 +244,35 @@ object Model {
...
-    case object Stopping          extends SequenceState
     case object Idle              extends SequenceState
-    case object Paused            extends SequenceState
-    final case class Error(msg: String) extends SequenceState
+    case object Stopped           extends SequenceState
+    final case class Running(userStop: Boolean, internalStop: Boolean) extends SequenceState
+    object Running {
+      val init: Running = Running(false, false)
+    }
+    final case class Failed(msg: String) extends SequenceState
```
Comment (jluhrs) [inline modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/Model.scala]: Changed to `Failed` to be consistent.

File: modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala (hunk 1 above)
Comment (cquiroz) [reply modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala]: Maybe these could be converted to `ops` via an implict class, but it's low prio

File: modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/Model.scala (hunk 2 above)
Comment (cquiroz) [reply modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/Model.scala]: 👍

**Pull request merged**